odxtools/table.py

```diff
@@ -59,19 +78,18 @@ def __init__(
         self,
         id: str,
         short_name: str,
-        key_dop_ref: str,
         table_rows: List[TableRow],
         long_name: Optional[str] = None,
+        key_dop_ref: Optional[str] = None,
     ):
-        super().__init__(
-            id=id, short_name=short_name, long_name=long_name, key_dop_ref=key_dop_ref
-        )
+        super().__init__(id=id, short_name=short_name, long_name=long_name)
         self.table_rows = table_rows
+        self.key_dop_ref = key_dop_ref
         self._key_dop = None
 
     @property
     def key_dop(self) -> Optional[DopBase]:
-        """The key data object property associated with this table."""
+        """The key_dop object resolved by key_dop_ref."""
```

Comment:
```suggestion
        """The key_dop object referenced by key_dop_ref."""
```

Reply:
This text is already updated by another MR by @kayoub5